spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -120,6 +120,11 @@ def _setup_matplotlib(self):
     # If we don't have matplotlib installed don't bother continuing
     try:
       import matplotlib
+
+      #matplotlib.rcParams['savefig.format'] is only avail in version 1.2.0 and above
+      from distutils.version import LooseVersion
+      if LooseVersion("1.2.0") >= LooseVersion(matplotlib.__version__):
+        return
```

Comment:
With this approach, the plotting backend setup is completely aborted if the matplotlib version is 1.2.0 or older. However this is bad because this also means matplotlib will not fallback to the Agg backend, which is essential since for most systems the default backend is not notebook compatible.

An alternative is to just catch a `KeyError`, but in general I think this is still not the proper solution to the problem since removing the custom backend over a single missing key is beyond overkill.